### PR TITLE
files: checksum verification

### PR DESCRIPTION
**Pull request by dinosk**
* ADD celery tasks to verify the checksum of stored files

Signed-off-by: Dinos Kousidis <konstantinos.kousidis@cern.ch>

### CODE REVIEW

File: b2share/modules/files/tasks.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+def failed_checksum_files_query():
+    return FileInstance.query.filter_by(last_check=None)
+
+
+@shared_task()
+def schedule_all_files_for_checksum():
+    current_app.config['FILES_REST_CHECKSUM_VERIFICATION_FILES_QUERY'] = \
+        'invenio_files_rest.tasks.default_checksum_verification_files_query'
+    schedule_checksum_verification.apply_async(kwargs={'batch_interval':
```
Comment (dinosk) [inline b2share/modules/files/tasks.py]: Sample parameters, to be tuned

**Comment by dinosk** [comment]
Remaining: setting the frequency of the verification tasks and adding them in celery beat schedule.

File: b2share/modules/files/tasks.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            '>' + str(result[0]) + '"</a>"' + \
+            '\n Record Id: {0}\n Bucket Id: {1}\n File key: {2}\n' + \
+            ' Path: {3}\n Old Checksum: {4}\n\n'.\
+            format(str(result[0]), str(result[1]),
+                   str(result[2]), str(result[3]),
+                   str(result[4]))
+    error_count = FileInstance.query.filter_by(last_check=None).count()
+
+    msg_content = '{0}: {1}\n{2}: {3}'.\
+        format(_('Possibly Corrupted Files'),
```
Comment (nharraud) [inline b2share/modules/files/tasks.py]: The files are not corrupted at that point. It is more probable that the filesystem is not reachable. We can just report "An error occurred while checking files checksum".

File: b2share/modules/files/tasks.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    ))
+
+
+@shared_task()
+def schedule_all_files_for_checksum():
+    current_app.config['FILES_REST_CHECKSUM_VERIFICATION_FILES_QUERY'] = \
+        'invenio_files_rest.tasks.default_checksum_verification_files_query'
+    schedule_checksum_verification.apply_async(kwargs={'batch_interval':
+                                                       {'minutes': 10},
+                                                       'max_count': 0,
```
Comment (nharraud) [inline b2share/modules/files/tasks.py]: We need this dict as a config variable.

File: b2share/modules/files/tasks.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                                                       'max_size': 0})
+
+
+@shared_task()
+def schedule_failed_checksum_files():
+    current_app.config['FILES_REST_CHECKSUM_VERIFICATION_FILES_QUERY'] = \
+        'b2share.modules.files.tasks.failed_checksum_files_query'
+    schedule_checksum_verification.apply_async(kwargs={'batch_interval':
+                                                       {'minutes': 10},
+                                                       'max_count': 0,
```
Comment (nharraud) [inline b2share/modules/files/tasks.py]: We need this dict as a config variable.

File: b2share/modules/files/tasks.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        subject=_('B2SHARE Checksum Verification Report'),
+        sender=support,
+        recipients=[support],
+        body=msg_content
+    ))
+
+
+@shared_task()
+def schedule_all_files_for_checksum():
+    current_app.config['FILES_REST_CHECKSUM_VERIFICATION_FILES_QUERY'] = \
```
Comment (nharraud) [inline b2share/modules/files/tasks.py]: This doesn't work with Celery. Let's merge first inveniosoftware/invenio-files-rest#168 and use arguments.

File: tests/b2share_unit_tests/files/test_verify_checksum.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+from invenio_files_rest.tasks import verify_checksum
+
+from b2share_unit_tests.helpers import create_deposit, create_user, \
+    generate_record_data, url_for_file
+from b2share.modules.files.tasks import schedule_all_files_for_checksum, \
+    schedule_failed_checksum_files
+
+
+def test_verify_checksum(app, tmp_location):
+    """Test that verify_checksum sets last_check=False
```
Comment (nharraud) [inline tests/b2share_unit_tests/files/test_verify_checksum.py]: This docstring could be more explicit.
What about:
"""Test that the celery task verify_checksum detects wrong checksums."""

File: tests/b2share_unit_tests/files/test_verify_checksum.py [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+                as mock_check:
+            mock_check.side_effect = KeyError()
+            verify_checksum.apply_async(args=[str(failed_file.id)],
+                                        kwargs={'throws': False})
+            assert failed_file.last_check is None
+
+
+def test_verify_checksum_in_deposit(app, test_communities,
+                                    login_user, test_users):
+    """Test uploading and reading deposit files."""
```
Comment (nharraud) [inline tests/b2share_unit_tests/files/test_verify_checksum.py]: docstring doesn't match the test.

**Pull request merged**